onboarding_auth/onboardingAuthPresentation/src/main/java/com/alexafit/onboardingAuthPresentation/login/LoginViewModel.kt

```diff
@@ -110,4 +110,41 @@ class LoginViewModel @Inject constructor(
             }
         }
     }
+
+    private fun loginUser() {
+        viewModelScope.launch {
+            _uiEvent.send(
+                UiEvent.Loading(true)
+            )
+            onboardingAuthUseCase
+                .loginUser(
+                    email = loginState.emailAddress,
+                    passord = loginState.password
+                )
+                .onSuccess {
+                    /**
+                     * save token in room or datastore for later retrieval
+                     */
+                    _uiEvent.send(
+                        UiEvent.Loading(false)
+                    )
```

Comment:
Another thing I like to have, is 1 event class for each screen. Same with the LoginState, now LoginEvent

Mostly so you know exactly what each screen does, and dont call events that arent correctly handled and so on